homeassistant/components/samsungtv/bridge.py

```diff
@@ -0,0 +1,249 @@
+
+    def is_on(self):
+        """Tells if the TV is on."""
+        if self._remote is not None:
+            # Close the current remote connection
+            self._remote.close()
+            self._remote = None
+
+        try:
+            self._get_remote()
+            if self._remote:
+                return True
+        except (
+            UnhandledResponse,
+            AccessDenied,
+        ):
+            # We got a response so it's working.
+            return True
+        except OSError:
+            # Different reasons, e.g. hostname not resolveable
+            return False
+
+        return False
+
+    def _get_remote(self):
+        """Create or return a remote control instance."""
+        if self._remote is None:
+            # We need to create a new instance to reconnect.
+            try:
+                LOGGER.debug("Create SamsungRemote")
+                self._remote = Remote(self.config.copy())
+            # This is only happening when the auth was switched to DENY
+            # A removed auth will lead to socket timeout because waiting for auth popup is just an open socket
+            except AccessDenied:
+                self._notify_callback()
+                raise
+        return self._remote
+
+    def _send_key(self, key):
+        """Send the key using legacy protocol."""
+        self._get_remote().control(key)
+
+
+class SamsungTVWSBridge(SamsungTVBridge):
+    """The Bridge for WebSocket TVs."""
+
+    def __init__(self, config):
+        """Initialize Bridge."""
+        super().__init__(config)
+        self.config = config
+
+    def try_connect(self, port):
+        """Try to connect to the Websocket TV."""
+        for self.port in (8001, 8002):
+            if port is not None and port != self.port:
+                continue
+            config = {
+                "name": "HomeAssistant",
+                "description": "HomeAssistant",
+                "host": self.host,
+                "method": self.method,
+                "port": self.port,
+                # We need this high timeout because waiting for auth popup is just an open socket
+                "timeout": 31,
+                "token": self.token,
+            }
+            try:
+                LOGGER.debug("Try config: %s", config)
+                with SamsungTVWS(
+                    host=self.host,
+                    port=self.port,
+                    token=self.token,
+                    timeout=config["timeout"],
+                    name=config["name"],
+                ) as remote:
+                    remote.open()
+                LOGGER.debug("Working config: %s", config)
+                LOGGER.debug("Token: %s", self.token)
+                return RESULT_SUCCESS
+            except WebSocketException:
+                LOGGER.debug("Working but unsupported config: %s", config)
+                return RESULT_NOT_SUPPORTED
+            except (OSError, Exception) as err:  # pylint: disable=broad-except
```

Comment:
Do we need to except `Exception`? Can we narrow it down?

If we need to catch unexpected errors we should do that in a separate case and log that it's an unknown error.

Reply:
Samsungtvws raises only a base Exception... I'll make a PR there

Reply:
@xchwarze merged the PR, as soon as he will publish the updated version on pypi I'll adapt the code.

Reply:
PR has been accepted and new version released.